### PR TITLE
Backport fix for #1295 to 1.2.x

### PR DESCRIPTION
**Pull request by zregvart**
You might want to look at #1359 if it's merged first.

### CODE REVIEW

File: app/ui/src/app/integrations/edit-page/step-configure/filter-steps/basic-filter.component.ts [hunk 1]
```diff
@@ -115,8 +115,7 @@ export class BasicFilterComponent implements OnChanges {
     this.integrationSupport
       .getFilterOptions(dataShape)
       .toPromise()
-      .then((resp: any) => {
-        const body = JSON.parse(resp['_body']);
+      .then((body: any) => {
```
Comment (gashcrumb) [inline app/ui/src/app/integrations/edit-page/step-configure/filter-steps/basic-filter.component.ts]: You don't want this change on this branch, this version still uses the restangular impl.

Comment (rhuss) [reply app/ui/src/app/integrations/edit-page/step-configure/filter-steps/basic-filter.component.ts]: If this is fixed, I'm happy to merge (and kick of a new TP3 release)

**Comment by zregvart** [comment]
@gashcrumb @rhuss could you take a look, we've taken a bit different approach on `master` with #1374. Don't mind the Codacy issue, I can't seem to get it to analyze, though I'm sure it would find something.

File: app/connectors/connectors/sql/sql-common/pom.xml [hunk 2]
```diff
@@ -55,8 +55,9 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>org.json</groupId>
-      <artifactId>json</artifactId>
+      <groupId>com.vaadin.external.google</groupId>
+      <artifactId>android-json</artifactId>
+      <version>0.0.20131108.vaadin1</version>
```
Comment (rhuss) [inline app/connectors/connectors/sql/sql-common/pom.xml]: Why is this lib better ? Version number looks scary (major and datewise)

Comment (rhuss) [reply app/connectors/connectors/sql/sql-common/pom.xml]: BTW, I was and always am a big fan of json-simple (only dep of Jolokia)

Never let me down ...

Comment (zregvart) [reply app/connectors/connectors/sql/sql-common/pom.xml]: Not better, just under a different license, JSONAssert uses this implementation of `org.json` which is a clean room implementation there is a bit of text on this in 81102c278ad47c76e2f71c6a0fb750ad3d9a9edc. Without this we have duplicate classes on classpath.

Comment (rhuss) [reply app/connectors/connectors/sql/sql-common/pom.xml]: JSONAssert is only for testing, or ? Is it used also during runtime ? wonder why jsonassert dep is included not only for scope test.

**Comment by rhuss** [comment]
looks good to me. Just a small comment, dont know whether you want to address it. if not, just remove the wip label.

**Comment by rhuss** [comment]
@zregvart I just did a quick smoke test with this fix included in 1.2.8 but it still does not seem that the pull down works if there is Twitte before the filter step. If the filter step is after a datamapping step (with DB as target), then the autocomopletion works, though.

**Comment by zregvart** [comment]
I tested with DB step, could be that Java introspection is broken also.

**Pull request merged**